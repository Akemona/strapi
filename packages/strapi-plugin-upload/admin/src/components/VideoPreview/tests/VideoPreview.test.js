import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import VideoPreview from '..';
import themes from '../../../../../../strapi-admin/admin/src/themes';

jest.mock('react-intl', () => ({
  // eslint-disable-next-line react/prop-types
  FormattedMessage: ({ id }) => <div>{id}</div>,
  useIntl: () => ({
    formatMessage: ({ id }) => id,
  }),
}));

describe('VideoPreview', () => {
  it('shows its initial state with no props', () => {
    const { container } = render(
      <ThemeProvider theme={themes}>
        <VideoPreview />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it('shows a loading state when resolving the asset', () => {
    render(
      <ThemeProvider theme={themes}>
        <VideoPreview
          hasIcon
          previewUrl="https://some-preview-url/img.jpg"
          src="https://something-good/video.mp4"
        />
      </ThemeProvider>
    );

    expect(screen.getByLabelText('upload.list.assets.loading-asset')).toBeVisible();
  });

  it('shows the thumbnail but not the video when previewURL is passed', () => {
    const { container } = render(
      <ThemeProvider theme={themes}>
        <VideoPreview
          hasIcon
          previewUrl="https://some-preview-url/img.jpg"
          src="https://something-good/video.mp4"
        />
      </ThemeProvider>
    );

    expect(screen.getByAltText('upload.list.assets.preview-asset')).toBeVisible();

    expect(container.querySelector('video')).toBeFalsy();
  });

  it('shows the video when the previewURL is not passed', () => {
    const { container } = render(
      <ThemeProvider theme={themes}>
        <VideoPreview hasIcon src="https://something-good/video.mp4" />
      </ThemeProvider>
    );

    expect(container.querySelector('video')).toBeInTheDocument();
  });

  it('shows a fallback message when the video is in error', () => {
    const { container } = render(
      <ThemeProvider theme={themes}>
        <VideoPreview hasIcon src="https://something-good/video.wvf" />
      </ThemeProvider>
    );

    fireEvent(container.querySelector('video'), new Event('error'));

    expect(screen.getByText('upload.list.assets.not-supported-content')).toBeVisible();
  });
});
