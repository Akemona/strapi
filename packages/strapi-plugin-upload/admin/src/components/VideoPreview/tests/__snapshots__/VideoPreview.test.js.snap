// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPreview shows its initial state with no props 1`] = `
.c3 {
  margin: 0;
  line-height: normal;
  color: #ffffff;
  font-size: 13px;
  font-weight: 500;
  text-transform: none;
}

.c4 {
  position: absolute;
  bottom: 10px;
  right: 10px;
  padding: 3px 5px;
  border-radius: 2px;
  background-color: #333740;
}

.c1 {
  position: relative;
  width: 44%;
  height: 4px;
  overflow: hidden;
  background-color: #515764;
  border-radius: 2px;
}

.c1:before {
  content: '';
  display: block;
  position: absolute;
  left: -100px;
  width: 100px;
  height: 4px;
  background-color: #b3b5b9;
  -webkit-animation: gPeyzP 2s linear infinite;
  animation: gPeyzP 2s linear infinite;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 video {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c2 canvas {
  display: block;
  max-width: 100%;
  max-height: 100%;
  width: auto;
  height: auto;
  margin: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      aria-label="upload.list.assets.loading-asset"
      class="c1"
    />
    <div
      class="c2"
    >
      <video />
      <canvas />
      <p
        class="c3 c4"
        color="white"
        font-size="md"
        font-weight="semiBold"
      >
        00:00
      </p>
    </div>
  </div>
</div>
`;
